stable/ambassador/templates/daemonset.yaml

```diff
@@ -0,0 +1,146 @@
+        app.kubernetes.io/instance: {{ .Release.Name }}
+      {{- if .Values.podAnnotations }}
+      annotations:
+        {{- toYaml .Values.podAnnotations | nindent 8 }}
+      {{- end }}
+    spec:
+      serviceAccountName: {{ include "ambassador.serviceAccountName" . }}
+      volumes:
+        {{- if .Values.prometheusExporter.enabled }}
+        - name: stats-exporter-mapping-config
+          configMap:
+            name: '{{ template "ambassador.fullname" . }}-config'
+            items:
+            - key: exporterConfiguration
+              path: mapping-config.yaml
+        {{- end }}
+      {{- with .Values.volumes }}
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      containers:
+        {{- if .Values.prometheusExporter.enabled }}
+        - name: prometheus-exporter
+          image: "{{ .Values.prometheusExporter.image.repository }}:{{ .Values.prometheusExporter.image.tag }}"
+          ports:
+            - name: metrics
+              containerPort: 9102
+            - name: listener
+              containerPort: 8125
+          args:
+            - --statsd.listen-udp=:8125
+            - --web.listen-address=:9102
+            - --statsd.mapping-config=/statsd-exporter/mapping-config.yaml
+          volumeMounts:
+            - name: stats-exporter-mapping-config
+              mountPath: /statsd-exporter/
+              readOnly: true
+        {{- end }}
+        - name: {{ .Chart.Name }}
+          image: "{{ .Values.image.repository }}:{{ include "ambassador.imageTag" . }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          ports:
+            {{- if .Values.service.http.enabled }}
+            - name: http
+              containerPort: {{ .Values.service.http.targetPort }}
+            {{- end }}
+            {{- if .Values.service.https.enabled }}
+            - name: https
+              containerPort: {{ .Values.service.https.targetPort }}
+            {{- end }}
+            - name: admin
+              containerPort: 8877
+          env:
+            {{- if .Values.prometheusExporter.enabled }}
+            - name: STATSD_ENABLED
+              value: "true"
+            - name: STATSD_HOST
+              value: "localhost"
+            {{- end }}
+            {{- if .Values.namespace.single }}
+            - name: AMBASSADOR_SINGLE_NAMESPACE
+              value: {{ .Values.namespace.single | quote }}
+            {{- end }}
+            - name: AMBASSADOR_ID
+              value: {{ .Values.ambassador.id | quote }}
+            - name: AMBASSADOR_NAMESPACE
+              {{- if .Values.namespace.name }}
+              value: {{ .Values.namespace.name | quote }}
+              {{ else }}
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.namespace
+              {{- end -}}
+            {{- if .Values.timing }}
+            {{- if .Values.timing.restart }}
+            - name: AMBASSADOR_RESTART_TIME
```

Comment:
Why define these manually in Helm vs let user specify via them `env`?

Reply:
Custom envs were just added to the upstream chart. Where most things were kept for backwards compatibility. I think it makes sense to remove them and let them be specified with env since there's a bunch of options.